bin/MorphingMSSMvsSM.cpp

```diff
@@ -538,88 +613,106 @@ int main(int argc, char **argv) {
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, {"qqh"}, sm_and_btag_cats, true); // sm categories + b-tagged mssm categories
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, ch::JoinStr({main_sm_signals, sm_signals}), cats[chn], true);
-    }
-    else if(analysis == "mssm_vs_sm_h125"){
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, {"ggh"}, sm_and_btag_cats, true); // sm categories + b-tagged mssm categories
-      cb.AddProcesses(SUSYggH_masses[era], {"htt"}, {era_tag}, {chn}, {"ggH_i", "ggH_t", "ggH_b", "ggA_i", "ggA_t", "ggA_b"}, mssm_cats, true);
-      cb.AddProcesses(SUSYbbH_masses[era], {"htt"}, {era_tag}, {chn}, {"bbh"}, mssm_btag_cats, true); // b-tagged mssm categories
-      cb.AddProcesses(SUSYbbH_masses[era], {"htt"}, {era_tag}, {chn}, {"bbH", "bbA"}, mssm_cats, true); // high mass categories only (== all mssm categories)
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, {"qqh"}, sm_and_btag_cats, true); // sm categories + b-tagged mssm categories
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, ch::JoinStr({main_sm_signals, sm_signals}), cats[chn], true);
-    }
-    else if(analysis == "mssm_vs_sm_CPV"){
-      cb.AddProcesses(SUSYggH_masses[era], {"htt"}, {era_tag}, {chn}, mssm_ggH_signals, cats[chn], true);
-      cb.AddProcesses(SUSYbbH_masses[era], {"htt"}, {era_tag}, {chn}, mssm_bbH_signals, cats[chn], true);
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, ch::JoinStr({main_sm_signals, sm_signals}), cats[chn], true);
-      cb.AddProcesses({""}, {"htt"}, {era_tag}, {chn}, {"qqH1"}, cats[chn], true);
+    else if(analysis == "bsm-model-dep-additional" || analysis == "bsm-model-dep-full"){
+      // Adding at first the additional Higgs boson signals
+      Categories additional_higgses_cats;
+      if(categorization == "classic") // classic categories cover full mass range for additional Higgs signals
+      {
+         additional_higgses_cats = cats[chn];
+      }
+      else if(categorization == "with-sm-ml" || categorization == "sm-ml-only")
+      {
+         if(sub_analysis == "sm-like-light" || sub_analysis == "cpv") // in that case, the additional Higgs bosons are relatively heavy
+         {
+         additional_higgses_cats = mssm_cats;
+         }
+         else if(sub_analysis == "sm-like-heavy") // in that case, all additional Higgs bosons are relatively light (< 200 GeV) --> don't consider them in high mass no-btag categories
+         {
+         additional_higgses_cats = sm_and_btag_cats;
+         }
+      }
+      cb.AddProcesses(SUSYggH_masses[era], {"htt"}, {era_tag}, {chn}, mssm_ggH_signals_additional, additional_higgses_cats, true);
+      cb.AddProcesses(SUSYbbH_masses[era], {"htt"}, {era_tag}, {chn}, mssm_bbH_signals_additional, additional_higgses_cats, true);
+
+      if(analysis == "bsm-model-dep-full")
```

Comment:
My problem with code duplication was that I would need to change things in different multiple places, and this can lead to errors. Therefore I more for that solution.

I tried to extend it with more comments to improve readability, but I may try to make some more comments on that

Reply:
I would also say, dont use nested ifs, this is very hard to understand who did not do the implementation by himself

Reply:
Yeah... I'll add more detailed comments on what is happening

Reply:
I am not sure how much comments are helping with this but we can try.

Reply:
Done. I'm still in favour of the nested way to keep the code short & flexible.

And I also think the logic, how the things are done is clear in that particular case of nested if's, and hopefully now clearer with more comments on it.